### PR TITLE
Modularized functions for divertor_monoblock input file 

### LINKED ISSUE #332: Modularize divertor monoblock test/example
The divertor monoblock test is used as a primary example of how to run TMAP8.  The current version includes a lot code replication and magic numbers that could be reduced by modularization:
* [ ] add a function for the general time history of a plasma discharge and use in current functions
* [ ] introduce variables for the many parameters that define a plasma discharge, e.g. ramp time, discharge time, down time, max tritium flux, max heat flux, etc..

If this replaces the existing divertor monoblock input file, then the documentation will need to be updated appropriately.

### PR DESCRIPTION
**Pull request by katielygre**
Resolves #332 
Added a function for the general time history of a plasma discharge and for use in other functions.
Added variables to define the plasma discharge (ramp time, discharge time, maximum heat flux, maximum tritium flux, etc.)
Used variables to scale and shift the general time plasma function to match the original input file.

### CODE REVIEW

**Comment by katielygre** [comment]
I can't request reviews so I am tagging @simopier @gonuke. Thanks!

File: test/tests/divertor_monoblock/divertor_monoblock.i [hunk 1]
```diff
@@ -16,6 +16,18 @@
 
 tungsten_atomic_density = ${units 6.338e28 m^-3}
 
+plasma_ramp_time = 100.0
+plasma_ss_end_time = 500.0
+plasma_ramp_down_time = 600.0
+plasma_cycle_time = 1600.0
```
Comment (gonuke) [inline test/tests/divertor_monoblock/divertor_monoblock.i]: Minor comment: what if we defines these as durations and then add them to get the times - makes it easier to change just one of them and ensure the rest are consistent

File: test/tests/divertor_monoblock/divertor_monoblock.i [hunk 2]
```diff
@@ -683,18 +695,24 @@ tungsten_atomic_density = ${units 6.338e28 m^-3}
 [Functions]
     ### Maximum mobile flux of 7.90e-13 at the top surface (1.0e-4 [m])
     ### 1.80e23/m^2-s = (5.0e23/m^2-s) *(1-0.999) = (7.90e-13)*(${tungsten_atomic_density})/(1.0e-4)  at steady state
+    [plasma_time_func]
```
Comment (gonuke) [inline test/tests/divertor_monoblock/divertor_monoblock.i]: Maybe add another function just to calculate 
```
t % ${plasma_cycle_time}
```

File: test/tests/divertor_monoblock/divertor_monoblock.i [hunk 3]
```diff
@@ -683,18 +695,24 @@ tungsten_atomic_density = ${units 6.338e28 m^-3}
...
+                        if((t % ${plasma_cycle_time}) < ${plasma_ramp_down_time}, 1 - ((t % ${plasma_cycle_time})-${plasma_ss_end_time})/${plasma_ramp_time}, 0.0)))'
+    []
     [mobile_flux_bc_func]
         type = ParsedFunction
-        expression =   'if((t % 1600) < 100.0, 0.0      + 7.90e-13*(t % 1600)/100,
-                        if((t % 1600) < 500.0, 7.90e-13,
-                        if((t % 1600) < 600.0, 7.90e-13 - 7.90e-13*((t % 1600)-500)/100, 0.0)))'
+        symbol_values = 'plasma_time_func ${plasma_max_flux} ${plasma_min_flux}'
+        symbol_names = 'time_func max_val min_val'
+        expression = '(max_val - min_val) * time_func + min_val'
```
Comment (gonuke) [inline test/tests/divertor_monoblock/divertor_monoblock.i]: Wonder if this makes it simpler to track which function it is? (Assuming it works?)
```suggestion
        symbol_values = 'plasma_time_func'
        symbol_names = 'time_func'
        expression = '(${plasma_max_flux} - ${plasma_min_flux}) * time_func + ${plasma_min_flux}'
```
... and similarly for other cases

File: test/tests/divertor_monoblock/divertor_monoblock.i (hunk 1 above)
Comment (simopier) [reply test/tests/divertor_monoblock/divertor_monoblock.i]: I agree with this suggestion.

File: test/tests/divertor_monoblock/divertor_monoblock.i (hunk 3 above)
Comment (simopier) [reply test/tests/divertor_monoblock/divertor_monoblock.i]: Yes, this removes unnecessary intermediate steps/quantities/definitions.

File: test/tests/divertor_monoblock/divertor_monoblock.i (hunk 2 above)
Comment (simopier) [inline test/tests/divertor_monoblock/divertor_monoblock.i]:
```suggestion
    [plasma_time_function]
```
I prefer to avoid abbreviations. It is usually more confusing than anything else.

Comment (simopier) [reply test/tests/divertor_monoblock/divertor_monoblock.i]: I I know you are just being consistent with the other function names, but I'll update that in my clean up.

**Review by simopier — CHANGES_REQUESTED** [review]
Thank you @katielygre!

The tests currently fail because you need to reference the issue in one of your commit message. Something like `(Close #332)`. This is required for SQA reasons; it's not enough to reference the issue number in the description of the PR, you also need it in one of the commit message. 

This is why you are getting the pre-check error:
```
##########################################################################
ERROR: Your patch does not contain a valid ticket reference! (i.e. #1234)
Merge branch 'devel' of https://github.com/katielygre/TMAP8 into test
Merge branch 'idaholab:devel' into devel
Added timing functions for plasma discharge
##########################################################################
```

File: test/tests/divertor_monoblock/divertor_monoblock.i [hunk 4]
```diff
@@ -16,6 +16,18 @@
...
+plasma_ramp_time = 100.0
+plasma_ss_end_time = 500.0
+plasma_ramp_down_time = 600.0
+plasma_cycle_time = 1600.0
+
+plasma_max_heat = 1.0e7
+plasma_min_heat = 300.0
+
+plasma_max_flux = 7.90e-13
+plasma_min_flux = 0.0
```
[the remark this answers is not in the document]
Comment (katielygre) [reply test/tests/divertor_monoblock/divertor_monoblock.i]: What exactly are the units for this mobile flux boundary condition? I see it’s being equated to a physical flux, but I’m not fully sure how you’d like it defined in the input file. I know you might be planning to clarify this on your end, so I can leave it as is for now.

File: test/tests/divertor_monoblock/divertor_monoblock.i [hunk 5]
```diff
@@ -16,6 +16,21 @@
...
 
+plasma_ramp_time = 100.0 #s
+plasma_ss_duration = 400.0 #s
+plasma_cycle_time = 1600.0 #s
+
+plasma_ss_end = ${fparse plasma_ramp_time + plasma_ss_duration} #s
+plasma_ramp_down_end = ${fparse plasma_ramp_time + plasma_ss_duration + plasma_ramp_time} #s
+
+plasma_max_heat = 1.0e7 #W/m^2
+plasma_min_heat = 300.0 #W/m^2
```
Comment (simopier) [inline test/tests/divertor_monoblock/divertor_monoblock.i]:
```suggestion
plasma_min_heat = 0.0 # W/m^2 # no flux while the pulse is off. 
```

**Review by simopier — CHANGES_REQUESTED** [review]
I have additional comments for you, @katielygre, which you can find below. 

I also noticed that the heavy test for the divertor monoblock model was failing. You were not predicting the same behavior as before your changes. So I looked more into it and found the issue, which I described here (#339) and fix here (#340). We'll need to wait for this change to be merged before your changes can be merged. As we discuss, your PR should really not change the behavior of the tests. 

Also, if you have a bit of time, I would encourage you to do the same changes you do here in the input files `divertor_monoblock_physics-single-var.i` and `divertor_monoblock_physics.i`. I know I told you not to at first to keep things simple, but since you are handling this change well and will need to wait for the other PR to get in, you might as well do it all together here with my latest suggestions. 

Regarding suggestions, to facilitate incorporating these comments and the following reviews, I suggest going to the `Files changed` tab and clicking "add to batch" to the suggestions to agree with, and then go to the top of the page and commit all the suggestions. This will automatically close the suggested changes, and will facilitate follow up reviews. 

Let me know if you have any questions!

**Comment by simopier** [comment]
@katielygre https://github.com/idaholab/TMAP8/pull/340 is now merged, so you can now rebase, resolve conflict, and push this branch to update the PR. This should resolve the failing heavy tests.
Let me know if you have any questions.

**Comment by simopier** [comment]
You are missing a reference to an issue number in a commit message. You should have `(Ref. #332)` in one of your commit's message. This is why you are failing pre-check right now.
